src/language/dafnyInstallation.ts

```diff
@@ -114,7 +114,13 @@ export class DafnyInstaller {
       await this.extractArchive(archive);
       await workspace.fs.delete(archive, { useTrash: false });
       this.writeStatus('Dafny installation completed');
-      return true;
+      if(os.type() === 'Darwin' && os.arch() !== 'x64') {
+        // Need to build from source and move all files from Binary/ to the current foldre
+        this.writeStatus('Found a non-supported architecture OSX:' + os.arch() + '. Going to install from source and replace the automated installation.\n');
```

Comment:
Prefer the use of format strings. Line-breaks are automatically appended (easier for OS interoperability):
```suggestion
        this.writeStatus(`Found a non-supported architecture OSX: ${os.arch()}. Going to install from source and replace the automated installation.`);
```